### PR TITLE
refactor: Upgrade bazel and move to MODULE.bazel

### PR DESCRIPTION
**Pull request by narendasan**
# Description

Fixes the issues with compiling with bazel directly and upgrades to the MODULE.bazel system 

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [x] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: MODULE.bazel [hunk 1]
```diff
@@ -1,44 +1,37 @@
...
+    commit = "2677e021e51d9f6a81dc5568f7c44c21a7ff7d07",
+    remote = "https://github.com/narendasan/rules_pkg",
 )
 
+local_repository = use_repo_rule("@bazel_tools//tools/build_defs/repo:local.bzl", "local_repository")
+
 # External dependency for torch_tensorrt if you already have precompiled binaries.
 local_repository(
     name = "torch_tensorrt",
     path = "/opt/conda/lib/python3.8/site-packages/torch_tensorrt",
```
Comment (zewenli98) [inline MODULE.bazel:30]: Should this still be `python3.8`?

Comment (narendasan) [reply MODULE.bazel:30]: I think we just use this for DLFW CI so @apbose might know

Comment (narendasan) [reply MODULE.bazel:30]: Local repositories still exist for MODULE.bazel, there is just now a bazel package repository for common packages that we use

**Review by peri044 — APPROVED** [review]
LGTM

File: packaging/pre_build_script_windows.sh [hunk 2]
```diff
@@ -8,11 +8,11 @@ pip install tensorrt==${TRT_VERSION} tensorrt-${CU_VERSION::4}-bindings==${TRT_V
 
 choco install bazelisk -y
 
-curl -Lo TensorRT.zip https://developer.nvidia.com/downloads/compute/machine-learning/tensorrt/10.1.0/zip/TensorRT-10.1.0.27.Windows.win10.cuda-12.4.zip
-unzip -o TensorRT.zip -d C:/
+#curl -Lo TensorRT.zip https://developer.download.nvidia.com/compute/machine-learning/tensorrt/10.0.1/zip/TensorRT-10.0.1.6.Windows10.win10.cuda-12.4.zip
```
Comment (zewenli98) [inline packaging/pre_build_script_windows.sh:11]: doesn't need to change to TRT 10.0 I think, although it's commented out.

**Review by zewenli98 — APPROVED** [review]
LGTM, just a minor comment

File: MODULE.bazel [hunk 3]
```diff
@@ -1,44 +1,37 @@
...
-http_archive(
-    name = "googletest",
-    sha256 = "755f9a39bc7205f5a0c428e920ddad092c33c8a1b46997def3f1d4a82aded6e1",
-    strip_prefix = "googletest-5ab508a01f9eb089207ee87fd547d290da39d015",
-    urls = ["https://github.com/google/googletest/archive/5ab508a01f9eb089207ee87fd547d290da39d015.zip"],
+bazel_dep(name = "rules_pkg", version = "1.0.1")
+git_override(
+    module_name = "rules_pkg",
+    commit = "17c57f4",
+    remote = "https://github.com/narendasan/rules_pkg",
```
[the remark this answers is not in the document]
Comment (narendasan) [reply MODULE.bazel:22]: Theres an issue with building in a root repo without the patches in my fork

**Pull request merged**